scripts/export_models_ops_correlation.py

```diff
@@ -0,0 +1,293 @@
+                    header=0,
+                    usecols=["OpName", "Operands Shape", "Attributes"],
+                )
+
+                if model_name not in models_ops_dict.keys():
+                    models_ops_dict[model_name] = op_config_df.OpName.tolist()
+                else:
+                    models_ops_dict[model_name].extend(op_config_df.OpName.tolist())
+
+    all_models_ops_list = []
+    for model, ops in models_ops_dict.items():
+        all_models_ops_list.append(pd.DataFrame({"Models": model, "Ops": ops}))
+
+    # Combine all into a single dataframe on index axis(i.e 0)
+    combined_models_ops_df = pd.concat(all_models_ops_list)
+
+    # Count occurrences of each operation across all models
+    op_count = combined_models_ops_df["Ops"].value_counts().rename("OpCount")
+
+    # Pivot to get Ops as rows and Models as columns
+    pivot_df = pd.pivot_table(
+        combined_models_ops_df,
+        index="Ops",
+        columns="Models",
+        aggfunc="size",
+        fill_value=0,
+    )
+
+    # Replace the counts with 'Y' for operations present, and 'N' for absent
+    pivot_df = pivot_df.applymap(lambda x: "Y" if x > 0 else "N")
+
+    # Add the operation counts as a new column
+    pivot_df["OpCount"] = op_count
+
+    # Sort rows by number of models supporting each operation
+    pivot_df["model_ops_count"] = pivot_df.apply(lambda row: (row == "Y").sum(), axis=1)
+    pivot_df = pivot_df.sort_values(by="model_ops_count", ascending=False).drop("model_ops_count", axis=1)
+
+    # Sort columns by model names in alphabetical order
+    column_names = list(pivot_df.columns)
+    column_names.remove("OpCount")
+    column_names.sort()
+    column_names.append("OpCount")
+    pivot_df = pivot_df[column_names]
+
+    if not do_save_xlsx:
+        pivot_df.to_csv(os.path.join(output_directory_path, output_file_name + ".csv"))
+        logger.info(
+            f"Saved cross correlation details between model variants vs unique op configs to {os.path.join(output_directory_path,output_file_name+'.csv')}"
+        )
+        return
+
+    # Create a new Excel writer object
+    writer = pd.ExcelWriter(
+        os.path.join(output_directory_path, output_file_name + ".xlsx"),
+        engine="openpyxl",
+    )
+
+    # Write the dataframe to the Excel file
+    pivot_df.to_excel(writer, sheet_name=output_file_name)
+
+    # Access the workbook and sheet
+    workbook = writer.book
+    ws = workbook.active
+    sheet = writer.sheets[output_file_name]
+
+    # Define your styles for fill the cell with particular color, center align the text in cell
+    green_fill = PatternFill(start_color="90EE90", end_color="90EE90", fill_type="solid")
+    black_fill = PatternFill(start_color="808080", end_color="808080", fill_type="solid")
+    blue_fill = PatternFill(start_color="6495ED", end_color="6495ED", fill_type="solid")
+    red_fill = PatternFill(start_color="FF7F7F", end_color="FF7F7F", fill_type="solid")
```

Comment:
Can we use red_fill for "N" as well? And use a neutral color for op appearance?

Reply:
Sure

Reply:
Updated